src/Planner/Planner.cpp

```diff
@@ -1279,10 +1326,24 @@ void addLimitStep(QueryPlan & query_plan,
 
     UInt64 limit_length = query_analysis_result.limit_length;
     UInt64 limit_offset = query_analysis_result.limit_offset;
+    Float32 fractional_limit = query_analysis_result.fractional_limit;
+    Float32 fractional_offset = query_analysis_result.fractional_offset;
     bool is_limit_length_negative = query_analysis_result.is_limit_length_negative;
     bool is_limit_offset_negative = query_analysis_result.is_limit_offset_negative;
 
-    if (!is_limit_length_negative && !is_limit_offset_negative) [[likely]]
+    if (is_limit_length_negative && fractional_offset > 0)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Negative Limits can't have a fractional Offset");
```

Comment:
```suggestion
        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Negative LIMIT is not supported with fractional OFFSET");
```

Maybe in the future, we will...